src/hotspot/share/gc/shenandoah/shenandoahMmuTracker.cpp

```diff
@@ -94,12 +94,17 @@ void ShenandoahMmuTracker::report() {
   // This is only called by the periodic thread.
   double process_time_s = process_time_seconds();
   double elapsed_process_time_s = process_time_s - _process_reference_time_s;
+  if (elapsed_process_time_s <= 0.01) {
```

Comment:
Is there a different mechanism one could use to check if there was an update? (like a count of GC cycles started or ended, rather than relying on the magnitude of the difference, although the magnitude of the difference may still be sufficient. One related question: is this unit "seconds" or some smaller unit? The `_s` implies it's a second and 0.01 s is 10 ms which is not insubstantial...)

Reply:
In general the synchronous, time-based, sampling of the MMU suffers from this issue of high variance by catching the co-initial or co-terminal portion of a cycle. Have you looked at the variance of MMUs to see if you should sample only at the end of complete cycles `synchronous with cycles` rather than using a `time-based sampling trigger`? Anyway something to think about and look into.

Reply:
I didn't want to use GC count changing because then it would miss 100% MMU (which is something to celebrate). This should be called once every GCPauseIntervalMillis (5000 by default). It's hard to imagine the process didn't have any CPU time over such an interval, but not impossible. Seems more likely the scheduled task thread fell behind and executed the task twice in quick succession. I'm not totally sure how this happens.